src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -104,512 +158,931 @@ enum Spells
-    {
-        return GetTrialOfTheChampionAI<boss_paletressAI>(creature);
-    }
 };
 
-class npc_memory : public CreatureScript
+class npc_fountain_of_light : public CreatureScript
 {
-public:
-    npc_memory() : CreatureScript("npc_memory") { }
+    public:
+        npc_fountain_of_light() : CreatureScript("npc_fountain_of_light") { }
 
-    struct npc_memoryAI : public ScriptedAI
-    {
-        npc_memoryAI(Creature* creature) : ScriptedAI(creature)
+        struct npc_fountain_of_lightAI : public ScriptedAI
         {
-            Initialize();
-        }
+            npc_fountain_of_lightAI(Creature* creature) : ScriptedAI(creature)
+            {
+                me->SetReactState(REACT_PASSIVE);
+                SetCombatMovement(false);
+            }
 
-        void Initialize()
-        {
-            uiOldWoundsTimer = 12000;
-            uiShadowPastTimer = 5000;
-            uiWakingNightmare = 7000;
-        }
+            void JustDied(Unit* /*killer*/) override
+            {
+                me->DespawnOrUnsummon();
+            }
+
+            void AttackStart(Unit* /*who*/) override { }
 
-        uint32 uiOldWoundsTimer;
-        uint32 uiShadowPastTimer;
-        uint32 uiWakingNightmare;
+            void UpdateAI(uint32 /*diff*/) override { }
+        };
 
-        void Reset() override
+        CreatureAI* GetAI(Creature* creature) const override
         {
-            Initialize();
+            return GetTrialOfChampionAI<npc_fountain_of_lightAI>(creature);
         }
+};
 
-        void UpdateAI(uint32 uiDiff) override
+class npc_memory : public CreatureScript
+{
+    public:
+        npc_memory() : CreatureScript("npc_memory") { }
+
+        struct npc_memoryAI : public ScriptedAI
         {
-            if (!UpdateVictim())
-                return;
+            npc_memoryAI(Creature* creature) : ScriptedAI(creature)
+            {
+                Initialize();
+				instance = creature->GetInstanceScript();
+                me->SetReactState(REACT_PASSIVE);
+            }
+			
+			InstanceScript* instance;
+			
+            void Initialize()
+            {
+                DoCastAOE(SPELL_SHADOWFORM);
+                DoCastAOE(SPELL_SPAWN_VISUAL);
+                _events.ScheduleEvent(EVENT_ENTER_AGGRESSIVE, 3 * IN_MILLISECONDS);
+            }
 
-            if (uiOldWoundsTimer <= uiDiff)
+            void Reset() override
             {
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
-                {
-                    if (target->IsAlive())
-                        DoCast(target, SPELL_OLD_WOUNDS);
-                }
-                uiOldWoundsTimer = 12000;
-            }else uiOldWoundsTimer -= uiDiff;
+                _events.Reset();
+                Initialize();
+            }
+
+            void JustDied(Unit* /*killer*/) override
+            {
+                /*if (Unit* summoner = me->ToTempSummon()->GetSummoner())
```

Comment:
Use `SummonedCreatureDies` on Confessor's AI.

Reply:
@Treeston Done - Thanks for advice :)